src/UsersModule/Components/AllRooms/AllRooms.tsx

```diff
@@ -1,12 +1,153 @@
-import React from 'react'
-import { useNavigate } from 'react-router-dom';
+import { Container, Typography } from '@mui/material';
+import { baseUrl } from '../../../Constants/Components/Urls';
+import React,{useEffect,useState} from 'react';
+import { json, useNavigate } from 'react-router-dom';
+import Link from '@mui/material/Link';
+import axios from 'axios';
+import Box from '@mui/material/Box';
+import Paper from '@mui/material/Paper';
+import ImageList from '@mui/material/ImageList';
+import ImageListItem from '@mui/material/ImageListItem';
+import NoData from '../../../SharedModule/Components/NoData/NoData';
+import { toast } from "react-toastify";
+import FavoriteIcon from '@mui/icons-material/Favorite';
+import VisibilityIcon from '@mui/icons-material/Visibility';
+import style from './AllRooms.module.css';
+import AnonymousUserAlert from '../AnonymousUserAlert/AnonymousUserAlert';
+
+
+
+
+
+
+
 
 export default function AllRooms() {
-    const navigate=useNavigate();
+  
+  let userRole=JSON.parse(localStorage.getItem('userRole')) ;
```

Comment:
This code will run in every render, it is not efficient since calling the localStorage is an expensive operation that we shouldn't be doing in every render.